vector/int.go

```diff
@@ -18,6 +18,14 @@ func NewInt(typ zed.Type, values []int64, nulls *Bool) *Int {
 	return &Int{Typ: typ, Values: values, Nulls: nulls}
 }
 
+func NewIntEmpty(typ zed.Type, length uint32, nulls *Bool) *Int {
+	return NewInt(typ, make([]int64, 0, length), nulls)
+}
+
+func (i *Int) Append(v int64) {
+	i.Values = append(i.Values, v)
```

Comment:
Do nulls need to be append to here?

Reply:
No need. The model for vector operations whose output is a fixed-size primitive type (like `Int`, `Uint`, `Float`, and `Bool`) is that we compute a value for every slot in the output vector regardless of nulls in the input vectors and separately compute an output nulls vector (the `Nulls *Bool` field present in most of the vector types) from the input nulls vectors.

This PR doesn't include that nulls handling and neither does #5146 or any of the existing code under runtime/vam/expr. But it's coming.